upup/pkg/fi/vfs_castore.go

```diff
@@ -153,59 +156,173 @@ func (c *VFSCAStore) generateCACertificate(id string) (*keyset, *keyset, error)
 	}
 
 	t := time.Now().UnixNano()
-	serial := pki.BuildPKISerial(t)
+	serial := pki.BuildPKISerial(t).String()
```

Comment:
Nit pick: Anyway that this returns a nil? `pki.BuildPKISerial(t) == nil`

Reply:
Good spot; thankfully it can't return nil (today).